ui/modals/neetcode.py

```diff
@@ -0,0 +1,28 @@
+from typing import TYPE_CHECKING
+
+import discord
+
+from ui.embeds.problems import neetcode_search_embed
+
+if TYPE_CHECKING:
+    # To prevent circular imports
+    from bot import DiscordBot
+
+
+class NeetcodeSearchModal(discord.ui.Modal, title="Search for a LeetCode problem"):
+    search_query_answer = discord.ui.TextInput(
+        label="Enter problem name, ID, or URL:",
+        style=discord.TextStyle.short,
+        required=True,
+    )
+
+    def __init__(self, bot: "DiscordBot") -> None:
+        self.bot = bot
+        super().__init__()
+
+    async def on_submit(self, interaction: discord.Interaction) -> None:
+        await interaction.response.defer()
+
+        embed = await neetcode_search_embed(self.bot, self.search_query_answer.value)
+
+        await interaction.followup.send(embed=embed)
```

Comment:
Rather than creating a new NeetCodeSearchModal, similar to ProblemSearchModal, let's just make ProblemSearchModal more general. We can do this by add a new param to the constructor of ProblemSearchModal called "submit_embed_creator" or something along those lines, and store it as an attribute. You can then replace the `search_question_embed` in `on_submit` with the `self.submit_embed_creator`, and then we can just pass the appropriate embed creator when we instantiate ProblemSearchModal.

So this allows us to pass the embed creator we want (e.g. search_question_embed with neetcode_search_embed (which I'd rename to search_neetcode_embed), and only have one modal class rather than two similar ones.